python/shotgun_model/data_handler.py

```diff
@@ -0,0 +1,473 @@
+# Copyright (c) 2016 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+from __future__ import with_statement
+
+import urlparse
+import errno
+import os
+import datetime
+import cPickle
+import time
+
+# toolkit imports
+import sgtk
+from sgtk.platform.qt import QtCore, QtGui
+
+from .errors import ShotgunModelDataError
+
+
+def log_timing(func):
+    """
+    Decorator that times and logs the execution of a method.
+    Borrowed from 0.18.
+    """
+    def wrapper(self, *args, **kwargs):
+        time_before = time.time()
+        try:
+            response = func(self, *args, **kwargs)
+        finally:
+            time_spent = time.time() - time_before
+            # log to special timing logger
+            self._bundle.log_debug("ShotgunDataHandler.%s took %fs" % (func.__name__, time_spent))
+        return response
+    return wrapper
+
+
+class ShotgunDataHandler(QtCore.QObject):
```

Comment:
hmm, that's a good question. I think at some point it emitted signals. But i don't think it does anymore. I'll see if i can just derive from object.